example/components/ConnectWithSelect.tsx

```diff
@@ -52,9 +52,19 @@ export function ConnectWithSelect({
   error: Error | undefined
   setError: (error: Error | undefined) => void
 }) {
+  const chainIds = (() => {
+    switch (true) {
+      case connector instanceof Network:
+        return Object.keys(URLS)
+      case connector instanceof WalletConnect:
+        return Object.keys(MAINNET_CHAINS)
```

Comment:
It does, but in order to make it work, you can't combine both test and main-nets together, so decided to do this

Reply:
Yeah, I guess we could have another card, but would require some refactoring to the cards itself (rather than having this switch, the chains should come as a parameter to the component). With the upcoming required/optional chains, this will require further tweaking anyway, so that we can incorporate things such as checkboxes etc.

Reply:
Update:
This is not implemented as a part of this PR and would require a bit of refactoring to make it happen. Right now, connector is defined statically outside of React context and you can't change once defined chains. I am definitely keeping that as a TODO for a follow-up, because we will need to deactivate/activate multiple occurrences of wallet connect connector in order to provide a nice playground for optional/required chains. For now, I am keeping it as is.